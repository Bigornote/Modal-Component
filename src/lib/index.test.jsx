import renderer from "react-test-renderer";
import { describe, expect, test, vi } from "vitest";
import Modal from "./index";

describe("Modal component", () => {
  test("renders correctly", () => {
    const modal = renderer.create(<Modal />);
    expect(modal.toJSON()).toMatchSnapshot();
  });

  test("renders title and message correctly", () => {
    const title = "Example Title";
    const message = "Example Message";
    const modal = renderer.create(<Modal title={title} message={message} />);
    const titleNode = modal.root.findByType("h3");
    const messageNode = modal.root.findByType("p");
    expect(titleNode.children).toContain(title);
    expect(messageNode.children).toContain(message);
  });

  test("calls closeModal prop on button click", () => {
    const closeModal = vi.fn();
    const modal = renderer.create(<Modal closeModal={closeModal} />);
    const closeButton = modal.root.findByProps({ className: "close-btn" });
    closeButton.props.onClick();
    expect(closeModal).toHaveBeenCalledTimes(1);
  });
});
