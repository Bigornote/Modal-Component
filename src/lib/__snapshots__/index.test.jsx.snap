// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal component > renders correctly 1`] = `
<div
  className="modal"
>
  <button
    className="close-btn"
  >
    <img
      alt="Close Modal Btn"
      src="/src/lib/assets/close.svg"
    />
  </button>
  <div
    className="img-wrapper"
    style={
      {
        "display": "none",
      }
    }
  >
    <img />
  </div>
  <div
    className="content-wrapper"
  >
    <h3 />
    <p />
  </div>
</div>
`;
